runtime/src/rpc.rs

```diff
@@ -70,12 +72,30 @@ impl PolkaBtcProvider {
 
     /// Gets a copy of the signer with a unique nonce
     async fn get_unique_signer(&self) -> PolkaBtcSigner {
-        // TODO: refresh from account store
         let mut signer = self.signer.write().await;
-        // return the current value, increment afterwards
-        let ret = signer.clone();
-        signer.increment_nonce();
-        ret
+        let mut account_nonce = self.nonce.load(Ordering::SeqCst);
+
+        // For getting the nonce, use latest, possibly non-finalized block.
+        // TODO: we might want to wait until the latest block is actually finalized
+        // query account info in order to get the nonce value used for communication
+        let system_nonce = crate::frame_system::AccountStoreExt::account(
```

Comment:
This is significant overhead, can we not get away with not doing this, and instead doing this only if a transaction fails due to the nonce being outdated? Presumably this shouldn't happen very often (only when the user is making transactions from the UI, right?)

Reply:
I don't think think the overhead will be too high, but in any case I've adapted this PR to only refresh on error.

Reply:
Cool, thanks